nifi-extension-bundles/nifi-extension-utils/nifi-record-utils/nifi-mock-record-utils/src/main/java/org/apache/nifi/serialization/record/MockSchemaRegistry.java

```diff
@@ -33,40 +33,59 @@
 import java.util.concurrent.ConcurrentMap;
 
 public class MockSchemaRegistry extends AbstractControllerService implements SchemaRegistry {
-    private final ConcurrentMap<String, RecordSchema> schemaNameMap = new ConcurrentHashMap<>();
+    private final ConcurrentMap<Triple<String, String, Integer>, RecordSchema> schemaNameMap = new ConcurrentHashMap<>();
     private final ConcurrentMap<Tuple<Long, Integer>, RecordSchema> schemaIdVersionMap = new ConcurrentHashMap<>();
 
     public void addSchema(final String name, final RecordSchema schema) {
-        schemaNameMap.put(name, schema);
+        addSchema(name, null, null, schema);
     }
 
-    RecordSchema retrieveSchemaByName(final SchemaIdentifier schemaIdentifier) throws IOException, SchemaNotFoundException {
+    public void addSchema(final String name, final String branch, final RecordSchema schema) {
+        addSchema(name, branch, null, schema);
+    }
+
+    public void addSchema(final String name, final Integer version, final RecordSchema schema) {
+        addSchema(name, null, version, schema);
+    }
+
+    public void addSchema(final String name, final String branch, final Integer version, final RecordSchema schema) {
+        schemaNameMap.put(Triple.of(name, branch, version), schema);
+    }
+
+    RecordSchema retrieveSchemaByName(final SchemaIdentifier schemaIdentifier) throws SchemaNotFoundException {
         final Optional<String> schemaName = schemaIdentifier.getName();
-        if (!schemaName.isPresent()) {
-            throw new org.apache.nifi.schema.access.SchemaNotFoundException("Cannot retrieve schema because Schema Name is not present");
+        if (schemaName.isEmpty()) {
+            throw new org.apache.nifi.schema.access.SchemaNotFoundException("Cannot retrieve schema because Schema Name is empty");
```

Comment:
I think the refactor to check for `isEmpty()` is easier to read than the inverted `isPresent()`.

However, I wouldn't change the exception message here. To me "is empty" on a String type sounds like it was an actual empty String such as `""` instead of a value being missing / `null` like here.

Reply:
Should be good.

Reply:
Thanks for the adjustments. 👍🏻
However, you seem to have missed this particular place. 

I wouldn't incorporate the "is empty" portion in the exception messages at all. I'd prefer if they either were unchanged from the original and say "is not present" or as you've worded it "is null".